### PR TITLE
Add pre commit action

### PR DESCRIPTION
**Pull request by phobson**
Needs #27 to go in first

### CODE REVIEW

File: crick/tests/test_tdigest.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from __future__ import print_function, division, absolute_import
+from __future__ import absolute_import, division, print_function
```
Comment (jrbourbeau) [inline crick/tests/test_tdigest.py:1]:
```suggestion
```

File: crick/tests/test_space_saving.py [hunk 2]
```diff
@@ -1,18 +1,17 @@
-from __future__ import print_function, division, absolute_import
+from __future__ import absolute_import, division, print_function
```
Comment (jrbourbeau) [inline crick/tests/test_space_saving.py:1]:
```suggestion
```

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: Linting
+
+on: [push]
```
Comment (jrbourbeau) [inline .github/workflows/pre-commit.yml:3]: Will this still appear as a PR check? Do we need to include `pull_request` here?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phobson

File: .github/workflows/pre-commit.yml (hunk 3 above)
Comment (phobson) [reply .github/workflows/pre-commit.yml:3]: Oh, we might. Good catch

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**